python/cuspatial/cuspatial/geometry/geoseries.py

```diff
@@ -0,0 +1,599 @@
+
+
+class GeoSeries(ColumnBase):
+    def __init__(self, data, name=None, index=None):
+        """
+        A GPU GeoSeries object.
+
+        Parameters
+        ----------
+        data : A GeoPandas GeoSeries object, or a file path
+
+        Discussion
+        ----------
+        The GeoArrow format specifies a tabular data format for geometry
+        information. Supported types include Point, MultiPoint, LineString,
+        MultiLineString, Polygon, and MultiPolygon.  In order to store
+        these coordinate types in a strictly tabular fashion, columns are
+        created for Points, MultiPoints, LineStrings, and Polygons.
+        MultiLines and MultiPolygons are stored in the same data structure
+        as LineStrings and Polygons.
+
+        The Points column is a simple numeric column. x and y coordinates
+        can be stored either interleaved or in separate columns. If a z
+        coordinate is present, it will be stored in a separate column.
+
+        The MultiPoints column is similar to the Points column with the
+        addition of an offsets column. The offsets column stores the comparable
+        sizes and coordinates of each MultiPoint in the cuGeoSeries.
+
+        LineStrings contain the coordinates column, an offsets column, and a
+        multioffsets column. The multioffsets column stores the indices of the
+        offsets that indicate the beginning and end of each MultiLineString
+        segment.
+
+        Polygons contain the coordinates column, a rings column specifying
+        the beginning and end of every polygon, a polygons column specifying
+        the beginning, or exterior, ring of each polygon and the end ring.
+        All rings after the first ring are interior rings.  Finally a
+        multipolys column stores the offsets of the polygons that should be
+        grouped into MultiPolygons.
+
+        As a result, a GpuGeoSeries object contains these arrow-supported
+        columns:
+        points_xy
+        mpoints_xy
+        mpoints_offsets
+        lines_xy
+        lines_offsets
+        lines_ml_offsets
+        polygons_xy
+        polygons_rings
+        polygons_polys
+        polygons_mpolys
+
+        Notes
+        -----
+        Legacy cuspatial algorithms depend on separated x and y columns.
+        De-interleave x and y columns with `column.de_interleave`, which
+        will create a legacy cuspatial Series, which violates the GeoArrow
+        format but will work with existing cuspatial algorithms.
+        """
+        if isinstance(data, GeoSeries):
+            self._data = data.data
+            self._points = data.points
+            self._multipoints = data.multipoints
+            self._lines = data.lines
+            self._polygons = data.polygons
+            self.types = data.types
+            self.lengths = data.lengths
+            if index is not None:
+                self.index = index
+            else:
+                self.index = data.index
```

Comment:
If these are a `cudf.Index` I imagine we'd want it to be `self._index` and then do validation as needed via the property setter / getter methods?

Reply:
Done, tho I'm not sure what kind of validation I should be doing on the inbound `cudf.Index`.

Reply:
Now that I am inheriting from GeoSeries, I'm writing a variety of `Series` level tests to ensure same functionality. Marking this resolved.